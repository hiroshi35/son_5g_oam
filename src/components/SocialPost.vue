<template>
  <div class="row">
    <div class="col-md-10 post">
      <panel type="panel-white right-angular" :footer="true">
        <div slot="header">
          <div class="row">
            <div class="col-md-8 col-xs-8">
              <div class="media">
                <div class="media-left">
                  <a href="#">
                    <img class="media-object img-circle" :src="author.avatar" width="40px" height="40px">
                  </a>
                </div>
                <div class="media-body">
                  <strong><a href="#" v-text="author.name"></a></strong> {{title}}<br>
                  <small v-text="date"><i class="fa fa-clock-o" aria-hidden="true"></i></small>
                </div>
              </div>
            </div>
            <div class="col-md-4 col-xs-4 config">
              <div class="dropdown pull-right">
                <button class="btn btn-config dropdown-toggle" type="button" :id="'postconfig' + id" data-toggle="dropdown" aria-haspopup="true" aria-expanded="true">
                  <i class="fa fa-chevron-down" aria-hidden="true"></i>
                </button>
                <ul class="dropdown-menu contextual-dropdown" :aria-labelledby="'postconfig' + id">
                  <li><a href="#"><i class="fa fa-times" aria-hidden="true"></i> Hide Post</a></li>
                  <li><a href="#"><i class="fa fa-reply" aria-hidden="true"></i> ReportPost</a></li>
                  <li><a href="#"><i class="fa fa-user-times" aria-hidden="true"></i>Unfollow</a></li>
                </ul>
              </div>
            </div>
          </div>
        </div>
        <div class="post-body">
          <slot></slot>
        </div>
        <div class="post-comments">
          <div class="col-md-12">
            <div class="divider"></div>
            <div class="comment" v-for="(comment, index) in comments">
              <div class="row">
                <div class="col-md-12 col-sm-12 col-xs-12">
                  <div class="media">
                    <div class="media-left">
                      <a href="#">
                        <img class="media-object img-circle" width="30px" :src="comment.author.avatar" height="30px">
                      </a>
                    </div>
                    <div class="media-body">
                      <p class="media-heading"><strong> <a href="#" v-html="comment.author.name"></a></strong>
                        <small class="pull-right"> {{comment.date}} <i class="fa fa-clock-o" aria-hidden="true"></i></small>
                      </p>
                      <p>{{comment.message}}</p>
                      <div>
                        <a href="#"> {{comment.likes}} <i class="fa fa-thumbs-o-up" aria-hidden="true"></i> </a> |
                        <a href="#">Like</a> |
                        <a href="#">Reply</a>
                      </div>
                    </div>

                  </div>
                </div>
              </div>
              <div class="divider" v-if="index + 1 != comments.length"></div>
            </div>
          </div>
        </div>
        <div slot="footer">
          <div class="row post-footer">
            <div class="col-md-4 col-sm-4 col-xs-4">
              <button class="btn btn-green rounded"><i class="fa fa-thumbs-o-up" aria-hidden="true"></i> {{likes}} Likes</button>
            </div>
            <div class="col-md-8 col-sm-8 col-xs-8">
              <form>
                <div class="input-group">
                  <input type="text" class="form-control rounded-left" placeholder="Write a Comment!">
                  <div class="input-group-btn">
                    <button type="button" class="btn btn-teal right-angular dropdown-toggle" data-toggle="dropdown"
                            aria-haspopup="true" aria-expanded="false">
                      <span class="caret"></span> <span class="sr-only">Toggle Dropdown</span>
                    </button>
                    <ul class="dropdown-menu contextual-dropdown dropdown-menu-right">
                      <li><a href="#"><i class="fa fa-camera" aria-hidden="true"></i> Add Photo</a></li>
                      <li><a href="#"><i class="fa fa-video-camera" aria-hidden="true"></i> Add Video</a></li>
                      <li><a href="#"><i class="fa fa-calendar-o" aria-hidden="true"></i> Add Event</a></li>
                    </ul>
                    <button type="button" class="btn btn-teal rounded-right">POST</button>
                  </div>
                </div>
              </form>
            </div>
          </div>
        </div>
      </panel>
    </div>
  </div>
</template>

<script>
  export default {
    props: {
      id: {
        type: Number,
        required: true
      },
      author: {
        type: Object,
        required: true
      },
      date: {
        type: String,
        required: true
      },
      title: {
        type: String,
        required: true
      },
      comments: {
        type: Array
      },
      likes: {
        type: Number,
        default: 0
      }
    }
  }
</script>