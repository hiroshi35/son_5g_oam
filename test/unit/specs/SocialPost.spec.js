import { mount } from 'vue-test-utils'
import SocialPost from '@/components/SocialPost.vue'

describe('SocialPost.vue', () => {
  let wrapper
  beforeEach(() => {
    wrapper = mount(SocialPost,
      {
        propsData: {
          id: 1,
          author: {
            name: 'John Smith ',
            avatar: '/static/imgs/avatars/man-1.png'
          },
          date: '4 hours ago',
          title: 'added a new photo ',
          likes: 15,
          body: '<p>Lorem ipsum dolor sit amet, consectetur adipisicing elit.</p><img src="/static/imgs/carousel/3.jpg" width="100%">',
          comments: [
            {
              author: {
                name: 'Frank Doe',
                avatar: '/static/imgs/avatars/man-4.png'
              },
              date: '3 hours ago',
              message: 'Lorem ipsum dolor sit amet, consectetur\n adipisicing elit. Corporis debitis dignissimos nisi obcaecati )',
              likes: 5
            },
            {
              author: {
                name: 'Sam Doe',
                avatar: '/static/imgs/avatars/man-7.png'
              },
              date: '2 hours ago',
              message: 'Lorem ipsum dolor sit amet, consectetur\n adipisicing elit. Corporis debitis dignissimos nisi obcaecati )',
              likes: 5
            }
          ]
        }
      })
  })
  it('should be vue instance', () => {
    expect(wrapper.isVueInstance()).toBeTruthy()
  })
  it('has the expected html structure', () => {
    expect(wrapper.element).toMatchSnapshot()
  })
})
