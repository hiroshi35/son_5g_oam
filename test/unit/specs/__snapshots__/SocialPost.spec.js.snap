// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SocialPost.vue has the expected html structure 1`] = `
<div
  class="row"
>
  <div
    class="col-md-10 post"
  >
    <panel
      footer="true"
      type="panel-white right-angular"
    >
      <div
        slot="header"
      >
        <div
          class="row"
        >
          <div
            class="col-md-8 col-xs-8"
          >
            <div
              class="media"
            >
              <div
                class="media-left"
              >
                <a
                  href="#"
                >
                  <img
                    class="media-object img-circle"
                    height="40px"
                    src="/static/imgs/avatars/man-1.png"
                    width="40px"
                  />
                </a>
              </div>
               
              <div
                class="media-body"
              >
                <strong>
                  <a
                    href="#"
                  >
                    John Smith 
                  </a>
                </strong>
                 added a new photo 
                <br />
                 
                <small>
                  4 hours ago
                </small>
              </div>
            </div>
          </div>
           
          <div
            class="col-md-4 col-xs-4 config"
          >
            <div
              class="dropdown pull-right"
            >
              <button
                aria-expanded="true"
                aria-haspopup="true"
                class="btn btn-config dropdown-toggle"
                data-toggle="dropdown"
                id="postconfig1"
                type="button"
              >
                <i
                  aria-hidden="true"
                  class="fa fa-chevron-down"
                />
              </button>
               
              <ul
                aria-labelledby="postconfig1"
                class="dropdown-menu contextual-dropdown"
              >
                <li>
                  <a
                    href="#"
                  >
                    <i
                      aria-hidden="true"
                      class="fa fa-times"
                    />
                     Hide Post
                  </a>
                </li>
                 
                <li>
                  <a
                    href="#"
                  >
                    <i
                      aria-hidden="true"
                      class="fa fa-reply"
                    />
                     ReportPost
                  </a>
                </li>
                 
                <li>
                  <a
                    href="#"
                  >
                    <i
                      aria-hidden="true"
                      class="fa fa-user-times"
                    />
                    Unfollow
                  </a>
                </li>
              </ul>
            </div>
          </div>
        </div>
      </div>
       
      <div
        class="post-body"
      />
       
      <div
        class="post-comments"
      >
        <div
          class="col-md-12"
        >
          <div
            class="divider"
          />
           
          <div
            class="comment"
          >
            <div
              class="row"
            >
              <div
                class="col-md-12 col-sm-12 col-xs-12"
              >
                <div
                  class="media"
                >
                  <div
                    class="media-left"
                  >
                    <a
                      href="#"
                    >
                      <img
                        class="media-object img-circle"
                        height="30px"
                        src="/static/imgs/avatars/man-4.png"
                        width="30px"
                      />
                    </a>
                  </div>
                   
                  <div
                    class="media-body"
                  >
                    <p
                      class="media-heading"
                    >
                      <strong>
                        <a
                          href="#"
                        >
                          Frank Doe
                        </a>
                      </strong>
                       
                      <small
                        class="pull-right"
                      >
                         3 hours ago 
                        <i
                          aria-hidden="true"
                          class="fa fa-clock-o"
                        />
                      </small>
                    </p>
                     
                    <p>
                      Lorem ipsum dolor sit amet, consectetur
 adipisicing elit. Corporis debitis dignissimos nisi obcaecati )
                    </p>
                     
                    <div>
                      <a
                        href="#"
                      >
                         5 
                        <i
                          aria-hidden="true"
                          class="fa fa-thumbs-o-up"
                        />
                      </a>
                       |
                      
                      <a
                        href="#"
                      >
                        Like
                      </a>
                       |
                      
                      <a
                        href="#"
                      >
                        Reply
                      </a>
                    </div>
                  </div>
                </div>
              </div>
            </div>
             
            <div
              class="divider"
            />
          </div>
          <div
            class="comment"
          >
            <div
              class="row"
            >
              <div
                class="col-md-12 col-sm-12 col-xs-12"
              >
                <div
                  class="media"
                >
                  <div
                    class="media-left"
                  >
                    <a
                      href="#"
                    >
                      <img
                        class="media-object img-circle"
                        height="30px"
                        src="/static/imgs/avatars/man-7.png"
                        width="30px"
                      />
                    </a>
                  </div>
                   
                  <div
                    class="media-body"
                  >
                    <p
                      class="media-heading"
                    >
                      <strong>
                        <a
                          href="#"
                        >
                          Sam Doe
                        </a>
                      </strong>
                       
                      <small
                        class="pull-right"
                      >
                         2 hours ago 
                        <i
                          aria-hidden="true"
                          class="fa fa-clock-o"
                        />
                      </small>
                    </p>
                     
                    <p>
                      Lorem ipsum dolor sit amet, consectetur
 adipisicing elit. Corporis debitis dignissimos nisi obcaecati )
                    </p>
                     
                    <div>
                      <a
                        href="#"
                      >
                         5 
                        <i
                          aria-hidden="true"
                          class="fa fa-thumbs-o-up"
                        />
                      </a>
                       |
                      
                      <a
                        href="#"
                      >
                        Like
                      </a>
                       |
                      
                      <a
                        href="#"
                      >
                        Reply
                      </a>
                    </div>
                  </div>
                </div>
              </div>
            </div>
             
            <!---->
          </div>
        </div>
      </div>
       
      <div
        slot="footer"
      >
        <div
          class="row post-footer"
        >
          <div
            class="col-md-4 col-sm-4 col-xs-4"
          >
            <button
              class="btn btn-green rounded"
            >
              <i
                aria-hidden="true"
                class="fa fa-thumbs-o-up"
              />
               15 Likes
            </button>
          </div>
           
          <div
            class="col-md-8 col-sm-8 col-xs-8"
          >
            <form>
              <div
                class="input-group"
              >
                <input
                  class="form-control rounded-left"
                  placeholder="Write a Comment!"
                  type="text"
                />
                 
                <div
                  class="input-group-btn"
                >
                  <button
                    aria-expanded="false"
                    aria-haspopup="true"
                    class="btn btn-teal right-angular dropdown-toggle"
                    data-toggle="dropdown"
                    type="button"
                  >
                    <span
                      class="caret"
                    />
                     
                    <span
                      class="sr-only"
                    >
                      Toggle Dropdown
                    </span>
                  </button>
                   
                  <ul
                    class="dropdown-menu contextual-dropdown dropdown-menu-right"
                  >
                    <li>
                      <a
                        href="#"
                      >
                        <i
                          aria-hidden="true"
                          class="fa fa-camera"
                        />
                         Add Photo
                      </a>
                    </li>
                     
                    <li>
                      <a
                        href="#"
                      >
                        <i
                          aria-hidden="true"
                          class="fa fa-video-camera"
                        />
                         Add Video
                      </a>
                    </li>
                     
                    <li>
                      <a
                        href="#"
                      >
                        <i
                          aria-hidden="true"
                          class="fa fa-calendar-o"
                        />
                         Add Event
                      </a>
                    </li>
                  </ul>
                   
                  <button
                    class="btn btn-teal rounded-right"
                    type="button"
                  >
                    POST
                  </button>
                </div>
              </div>
            </form>
          </div>
        </div>
      </div>
    </panel>
  </div>
</div>
`;
